test/unit/mongo_logger.test.ts

```diff
@@ -0,0 +1,347 @@
+          outcome: 'defaults to 1000',
+          expected: 1000
+        },
+        {
+          input: 'asdf',
+          context: 'when set to a non-integer string',
+          outcome: 'defaults to 1000',
+          expected: 1000
+        }
+      ];
+
+      for (const { input, outcome, expected, context: _context } of tests) {
+        context(_context, () => {
+          it(outcome, () => {
+            const options = MongoLogger.resolveOptions(
+              { MONGODB_LOG_MAX_DOCUMENT_LENGTH: input },
+              {}
+            );
+            expect(options.maxDocumentLength).to.equal(expected);
+          });
+        });
+      }
+    });
+
+    context('logDestination', function () {
+      const stream = new Writable();
+      const tests: Array<{
+        env: 'stderr' | 'stdout' | undefined;
+        client: MongoLoggerMongoClientOptions['mongodbLogPath'] | undefined;
+        expectedLogDestination: MongoLoggerOptions['logDestination'];
+      }> = [
+        {
+          env: undefined,
+          client: undefined,
+          expectedLogDestination: process.stderr
+        },
+        {
+          env: 'stderr',
+          client: undefined,
+          expectedLogDestination: process.stderr
+        },
+        {
+          env: 'stdout',
+          client: undefined,
+          expectedLogDestination: process.stdout
+        },
+        {
+          env: undefined,
+          client: 'stdout',
+          expectedLogDestination: process.stdout
+        },
+        {
+          env: 'stderr',
+          client: 'stdout',
+          expectedLogDestination: process.stdout
+        },
+        {
+          env: 'stdout',
+          client: 'stdout',
+          expectedLogDestination: process.stdout
+        },
+        {
+          env: undefined,
+          client: 'stderr',
+          expectedLogDestination: process.stderr
+        },
+        {
+          env: 'stderr',
+          client: 'stderr',
+          expectedLogDestination: process.stderr
+        },
+        {
+          env: 'stdout',
+          client: 'stderr',
+          expectedLogDestination: process.stderr
+        },
+        {
+          env: undefined,
+          client: stream,
+          expectedLogDestination: stream
+        },
+        {
+          env: 'stderr',
+          client: stream,
+          expectedLogDestination: stream
+        },
+        {
+          env: 'stdout',
+          client: stream,
+          expectedLogDestination: stream
+        }
+      ];
+
+      for (const { env, client, expectedLogDestination } of tests) {
+        context(
+          `environment option=${env}, client option=${
+            client instanceof Writable ? 'a writable stream' : client
+          }`,
+          () => {
+            it(`sets the log destination to ${
+              expectedLogDestination instanceof Writable
+                ? 'the provided writable stream'
+                : expectedLogDestination
```

Comment:
just a note: the expectedLogDestination is always a writable stream now

Reply:
fixed